src/parsers/perf/perfparser.cpp

```diff
@@ -1674,7 +1676,7 @@ void PerfParser::filterResults(const Data::FilterAction& filter)
         const bool excludeByBinary = !filter.excludeBinaries.isEmpty();
         const bool filterByStack = includeBySymbol || excludeBySymbol || includeByBinary || excludeByBinary;
 
-        if (!filter.isValid()) {
+        if (!filter.isValid() && !m_costAggregationChanged) {
```

Comment:
this is not threadsafe, you need to copy it into the lambda capture instead of accessing the bool directly here